test/DynamoCoreTests/UserDataMigrationTests.cs

```diff
@@ -0,0 +1,198 @@
+            fv2 = new FileVersion(0, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            
+            fv1 = new FileVersion(1, 7);
+            fv2 = new FileVersion(1, 8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(1, 9);
+            fv2 = new FileVersion(2, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-1, 9);
+            fv2 = new FileVersion(0, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, 9);
+            fv2 = new FileVersion(-1, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(3, 0);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(2, -9);
+            fv2 = new FileVersion(2, -8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-2, -8);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            fv1 = new FileVersion(-2, -9);
+            fv2 = new FileVersion(-1, -9);
+            list.Add(fv1);
+            list.Add(fv2);
+
+            list.Sort();
+
+            VerifySortedOrder(list);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void GetSortedInstalledVersions_FromDirectoryInspection()
+        {
+            var tempPath = Path.GetTempPath();
+            var uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "0.7"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "0.8"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "0.0"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "1.7"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "1.8"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "1.9"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "2.0"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-1.9"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-2.9"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-1.0"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "3.0"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "2.-9"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "2.-8"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-2.-9"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-2.-8"));
+            uniqueDirectory = Directory.CreateDirectory(Path.Combine(tempPath, "-1.-9"));
```

Comment:
This is crazily good, negative versions? How's that for negative tests, huh? :+1: